src/PluginInfo/PluginInfo.js

```diff
@@ -17,101 +17,112 @@
     under the License.
 */
 
-/*
-A class for holidng the information currently stored in plugin.xml
-It should also be able to answer questions like whether the plugin
-is compatible with a given engine version.
-
-TODO (kamrik): refactor this to not use sync functions and return promises.
-*/
-
 const path = require('path');
 const fs = require('fs-extra');
-const xml_helpers = require('../util/xml-helpers');
+const { parseElementtreeSync } = require('../util/xml-helpers');
 const CordovaError = require('../CordovaError');
 
+/**
+ * A class for holding the information currently stored in plugin.xml
+ *
+ * It should also be able to answer questions like whether the plugin
+ * is compatible with a given engine version.
+ */
 class PluginInfo {
     constructor (dirname) {
+        this.dir = dirname;
         this.filepath = path.join(dirname, 'plugin.xml');
+
         if (!fs.existsSync(this.filepath)) {
             throw new CordovaError(`Cannot find plugin.xml for plugin "${path.basename(dirname)}". Please try adding it again.`);
         }
 
-        this.dir = dirname;
-        const et = this._et = xml_helpers.parseElementtreeSync(this.filepath);
-        const pelem = et.getroot();
-        this.id = pelem.attrib.id;
-        this.version = pelem.attrib.version;
+        this._et = parseElementtreeSync(this.filepath);
+        const root = this._et.getroot();
+
+        this.id = root.attrib.id;
+        this.version = root.attrib.version;
 
         // Optional fields
-        this.name = pelem.findtext('name');
-        this.description = pelem.findtext('description');
-        this.license = pelem.findtext('license');
-        this.repo = pelem.findtext('repo');
-        this.issue = pelem.findtext('issue');
-        this.keywords = pelem.findtext('keywords');
-        this.info = pelem.findtext('info');
-        if (this.keywords) {
-            this.keywords = this.keywords.split(',').map(s => s.trim());
+        const optTags = 'name description license repo issue info'.split(' ');
```

Comment:
Can't this be an array? Does it really need to be string and then split on?

Reply:
This could be an array if desired. Although the string notation makes it less noisy and more compact IMHO.